src/odemis/acq/test/fastem_conf_test.py

```diff
@@ -0,0 +1,113 @@
+        cls.sed = model.getComponent(role="se-detector")
+        cls.multibeam = model.getComponent(role="multibeam")
+        cls.mppc = model.getComponent(role="mppc")
+
+    def test_configure_scanner(self):
+        """Check that for each available mode, the correct HW settings are set on the respective scanner VAs."""
+
+        # change the rotation for single and multibeam mode to check the correct one is selected depending on the mode
+        self.scanner.updateMetadata({model.MD_SINGLE_BEAM_ROTATION: math.radians(5),
+                                     model.MD_MULTI_BEAM_ROTATION: math.radians(10)})
+
+        # overview mode
+        fastem_conf.configure_scanner(self.scanner, OVERVIEW_MODE)
+
+        self.assertFalse(self.scanner.multiBeamMode.value)
+        self.assertFalse(self.scanner.external.value)
+        self.assertIsNone(self.scanner.blanker.value)
+        self.assertFalse(self.scanner.immersion.value)
+        self.assertGreater(self.scanner.horizontalFoV.value, 1.e-3)  # should be big FoV for overview
+        self.assertEqual(self.scanner.rotation.value, math.radians(5))
+
+        # check rotation set is also stored in MD as rotation correction
+        self.assertEqual(self.scanner.getMetadata()[model.MD_ROTATION_COR], math.radians(5))
+
+        # acquire an image and check the MD is correct: ROTATION - ROTATION_COR == 0
+        image = self.sed.data.get()
+        self.assertAlmostEqual(image.metadata[model.MD_ROTATION] - image.metadata[model.MD_ROTATION_COR], 0)
+        # merge the MD -> automatically calculates rotation - rotation_cor -> puts result on rotation in MD
+        img.mergeMetadata(image.metadata)
+        self.assertAlmostEqual(image.metadata[model.MD_ROTATION], 0)
+
+        # live mode
+        fastem_conf.configure_scanner(self.scanner, LIVESTREAM_MODE)
+
+        self.assertFalse(self.scanner.multiBeamMode.value)
+        self.assertFalse(self.scanner.external.value)
+        self.assertIsNone(self.scanner.blanker.value)
+        self.assertTrue(self.scanner.immersion.value)
+        self.assertEqual(self.scanner.rotation.value, math.radians(5))
+
+        # megafield mode
+        fastem_conf.configure_scanner(self.scanner, MEGAFIELD_MODE)
+
+        self.assertTrue(self.scanner.multiBeamMode.value)
+        self.assertTrue(self.scanner.external.value)
+        self.assertFalse(self.scanner.blanker.value)
+        self.assertTrue(self.scanner.immersion.value)
+        self.assertLess(self.scanner.horizontalFoV.value, 0.1e-3)  # should be smaller for megafield than for overview
+        self.assertEqual(self.scanner.rotation.value, math.radians(10))
+        
+        # check that rotation correction is 0 for megafield imaging using the ASM/SAM
+        self.assertEqual(self.multibeam.getMetadata().get(model.MD_ROTATION_COR, 0), 0)  # if set to 0 or not existing
+        self.assertEqual(self.multibeam.getMetadata()[model.MD_ROTATION], math.radians(10))  # same as on rotation VA
```

Comment:
I don't think these checks belong in `test_configure_scanner`, since you only configure the scanner not the multibeam.  Plus, if you remove them you can get rid of the asm component dependency.

Reply:
When you configure the scanner it affects the multibeam (set via affects in yaml file). As the affects was missing, we found out, that this MD was not updated, but it is essential that it gets updated. That's why this check is here.

Reply:
I agree that it's useful to check. However, Thera has a very interesting remark: that's the only place where the multibeam is needed. So when TEST_NOHW=1, instead of entirely giving up running the test cases, I'd suggest that we start the `FASTEM_CONFIG` (ie, the simulator without any external dependencies), and run everything *except* these 2 lines.

Reply:
The configure scanner method has some specific multibeam related VAs e.g. it sets the `scanner.multibeamMode.value` VA. This VA is not present when starting the current backend. So I changed the backend to have a `mb-scanner` child, which makes sure that the `MultiBeamScanner` is actually used. However, the `simsem` seems to need to have a `scanner ` child - otherwise the backend does not start. Any ideas on how to proceed with this?

Reply:
Arghhh.... I had forgotten about this. So the fastem-sim cannot be used at all to acquire any image in the GUI? That's a big limitation. For this PR, I think then you can leave it as is, but let's try to make fastem-sim more useful.

I think that we have two options:
* Extend SimSEM to simulate better the xt_client with multibeam (it's essentially a matter of adding a few VAs). Maybe via some flag (eg a parameter "multibeam" on the scanner), or the use of "mb-scanner" instead of "scanner", or a generic approach like `simulated.GenericComponent(..., vas...)`. It still doesn't bring any component related to the ASM.
* Bite the bullet and say that to run the fastem-sim you need to have the xtlib-adpater and ASM simulator running on your computer.